bridges/snowbridge/pallets/system/src/lib.rs

```diff
@@ -175,6 +198,8 @@ pub mod pallet {
 
 		#[cfg(feature = "runtime-benchmarks")]
 		type Helper: BenchmarkHelper<Self::RuntimeOrigin>;
+
+		type RegisterTokenOrigin: EnsureOrigin<Self::RuntimeOrigin, Success = AccountIdOf<Self>>;
```

Comment:
maybe

```suggestion
		type RegisterTokenOrigin: EnsureOrigin<Self::RuntimeOrigin, Success = Self::RuntimeOrigin>;
```

then you could:
```
let pays_fee = match origin {
	RawOrigin::Root => PaysFee::<T>::No,
	RawOrigin::Signed(who) => PaysFee::<T>::Yes(who),
```